### ISSUE TITLE
[feature] Toolchain lacks environment variables

### ISSUE THREAD

**Issue by ytimenkov** [conan-io/conan#7226]
The toolchain feature looks awesome and I was going to suggest it instead of `conan build --configure`, but it fails with 2 use-cases:
1. Profile defines compiler (e.g. `CXX=gcc-9`) and flags (`CXXFLAGS`) in environment variables. Those won't be used of CMake just runs.
2. If CMake itself is a Conan package, it won't be found.

Of course, one can use virtual(run)env generator in addition, but then configuring project will require 2 extra commands instead of one.

Besides, the command line for CMake is still complicated (toolchain, generator, path to sources - they are all known to Conan, why repeat).

I'm not sure if it's feasible to define environment variable (or compiler and flags) in the toolchain file as it addresses only the first use-case.

Another idea is to either generate an alias in virtual(run)env generator(s) (like `configure_cmake`) or conan subcommand (was also discussed iirc).

My goal is to give dead simple instructions to ones who don't want to master all details of build systems (like 90% of developers?). And avoid repetition.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
> Of course, one can use virtual(run)env generator in addition, but then configuring project will require 2 extra commands instead of one.

Yes, all things related to environment vars haven't been managed yet. It is on the TODO list, and it was delayed, because the approach is not very clear yet.

> Besides, the command line for CMake is still complicated (toolchain, generator, path to sources - they are all known to Conan, why repeat).

We would like that, but there are some issues. For example, CMake doesn't allow to define the generator in the toolchain, so it is still necessary to pass it in the command line, there is nothing else that can be done. Same with path to source. We are aiming to the minimal command line possible, but it doesn't seem possible to avoid those.

**Comment by Minimonium** [comment]
Related to #5737

**Comment by AbrilRBS** [comment]
Closed as solved back when in the final Conan 2 release and the finalization of the integrations
